OpenSim/Simulation/Model/ContactGeometry.h

```diff
@@ -33,120 +33,84 @@ namespace OpenSim {
-     */
-    OpenSim::PhysicalFrame& updBody();
-    /**
-     * %Set the Body this geometry is attached to.
-     */
-    void setBody(PhysicalFrame& body);
-    /**
-     * Get the name of the Body this geometry is attached to.
-     */
-    const std::string& getBodyName();
-    /**
-     * %Set the name of the Body this geometry is attached to.  This will cause the
-     * Body to be set to NULL, then resolved when extendConnectToModel() is called.
+    /** This constructor connects this ContactGeometry to the provided `frame`,
+     * and uses the default location and orientation (both `Vec3(0)`).
+     *
+     * @param frame        the PhysicalFrame this geometry is attached to;
      */
-    void setBodyName(const std::string& name);
+    explicit ContactGeometry(const PhysicalFrame& frame);
+
     /**
-     * Get the display_preference of this geometry.
+     * @param location     the location of the geometry expressed in `frame`
+     * @param orientation  the orientation of the geometry expressed in `frame`
+     *                     as XYZ body-fixed Euler angles.
+     * @param frame        the PhysicalFrame this geometry is attached to;
+     *                     this constructor connects this ContactGeometry to
+     *                     the provided `frame`
      */
+    ContactGeometry(const SimTK::Vec3& location,
+                    const SimTK::Vec3& orientation,
+                    const PhysicalFrame& frame);
+
+
+    // ACCESSORS
+    /** Get the PhysicalFrame this geometry is attached to. */
+    const PhysicalFrame& getFrame() const;
+    /** %Set the PhysicalFrame this geometry is attached to. */
+    void setFrame(const PhysicalFrame& body);
+
+    /** Get the display_preference of this geometry. */
     const int getDisplayPreference();
-    /**
-     * %Set the display_preference of this geometry.
-     */
+    /** %Set the display_preference of this geometry. */
     void setDisplayPreference(const int dispPref);
-    /**
-     * Create a new SimTK::ContactGeometry based on this object.
-     */
+
+    /** Create a new SimTK::ContactGeometry based on this object. */
     virtual SimTK::ContactGeometry createSimTKContactGeometry() = 0;
-    /**
-     * Get a Transform representing the position and orientation of the geometry
-     * within the Body it is attached to.
-     */
-    SimTK::Transform getTransform();
+
+    /** Get a Transform representing the position and orientation of the
+     * geometry within the base Frame (e.g., a Body or Ground) it is attached
+     * to (*not* the "frame" that the geometry is connected to). If `B` is the
+     * base or Body Frame, `F` is the PhysicalFrame that this geometry is
+     * connected to, and `P` is the frame that is defined (relative to `F`) by
+     * the location and orientation properties of this class, then this returns
+     * `X_BP = X_BF * X_FP`. */
+    SimTK::Transform findTransformInBaseFrame() const;
```

Comment:
I would prefer that the guts of the contact implementation become a tad more complex than introduce the concepts of `BaseFrame` or even that of a `MobilizedBody` frame to interface of ContactGeometry. ContactGeometry is conceptually complete with Geometry attached to a (Physical)Frame (with a build in convenience to be offset from that frame). Whoever consumes the `ContactGeometry` (e.g. `HuntCrossleyForce`) should drill down to a "Body" if/when it needs it.

I rather have the complexity in the implementation of the Force than in the interface of `ContactGeometry`. For example, I see nothing wrong with:

```cpp
contacts.addBody(set, geom.getFrame().getMobilizedBody(),
                     geom.createSimTKContactGeometry(),
                     geom.getFrame().findTransformInBaseFrame());
```
since you had to ask the frame to get its `MobilizedBody`  anyways.

Let's discuss it in person tomorrow.